### DISCUSSION TITLE
why not support Async Read/Write trait?

### DISCUSSION THREAD

**Discussion by R4cky** [b23r0/rust-raknet#6] [Q&A] [answered]
Why can't AsyncRead and AsyncWrite traits be supported like TcpStream

**Comment by b23r0** [comment, marked as answer]
Since Raknet is a message protocol, not a stream protocol like Tcp, it should implement features like UdpSocket.

According to the interface required by Async Read/Write, each IO needs to pass a fixed-size buffer to receive data, which may cause Raknet to fail to receive a complete or independent message each time IO.
